test/unit/validator.test.js

```diff
@@ -1076,3 +1076,233 @@ describe('VALIDATE FUNCTION TESTS ', function () {
     });
   });
 });
+
+describe('validateTransaction method. Path variables matching validation (issue #478)', function() {
+  it('Should validate correctly while a path param in spec does not matches with collection' +
+  ' (issue#478)', function(done) {
+    let issueFolder = path.join(__dirname, VALIDATION_DATA_FOLDER_PATH + '/issues/issue#478'),
+      issueSpec = fs.readFileSync(issueFolder + '/path-variable-does-not-match-spec.yaml', 'utf-8'),
+      issueCollection = fs.readFileSync(issueFolder + '/path-variable-does-not-match-collection.json', 'utf-8'),
+      resultObj,
+      historyRequest = [],
+      schemaPack = new Converter.SchemaPack({ type: 'string', data: issueSpec }, { allowUrlPathVarMatching: false });
+
+    getAllTransactions(JSON.parse(issueCollection), historyRequest);
+
+    schemaPack.validateTransaction(historyRequest, (err, result) => {
+      // Schema is sample petsore with one of parameter as empty, expect no mismatch / error
+      expect(err).to.be.null;
+      expect(result).to.be.an('object');
+      resultObj = result.requests[historyRequest[0].id].endpoints[0];
+      expect(resultObj.mismatches).to.have.length(1);
+      expect(resultObj.mismatches[0].reasonCode).to.be.equal('INVALID_VALUE');
+      expect(resultObj.mismatches[0].reason).to.be.equal(
```

Comment:
Can we also add test for verifying correct fix is suggested? i.e. `mismatches[0].suggestedFix` 's key, actualValue and suggestedValue. Also applies to other tests here.